### PR TITLE
Enable Wartremover

### LINKED ISSUE #24: Setup Wartremover
### Description

[Wartremover] looks like a nice tool. Let's try it out! It should perform the check on CI, of course. I think while the project is small it should be relatively easy to enforce any kind of linting and static checks.

[Wartremover]: https://github.com/puffnfresh/wartremover

### PR DESCRIPTION
**Pull request by rexim**
Close #24

### CODE REVIEW

**Comment by rexim** [comment]
I believe that we have to establish some policy about suppressing Warts. I'm gonna do that in the scope of this issue.

File: src/main/scala/me/rexim/issuestant/SplitIssueTree.scala [hunk 1]
```diff
@@ -5,9 +5,13 @@ object SplitIssueTree {
...
-  def print(level: Int = 0): Unit = {
+  private def printLevel(level: Int): Unit = {
     val padding = (1 to level).map(_ => ' ').mkString
     println(s"${padding}${number}: ${title}")
-    children.foreach(_.print(level + 1))
+    children.foreach(_.printLevel(level + 1))
+  }
+
+  def print(): Unit = {
+    printLevel(0)
```
Comment (ForNeVeR) [inline src/main/scala/me/rexim/issuestant/SplitIssueTree.scala]: What's the reason for this change? Are default parameters somewhat "unsafe"?

Comment (rexim) [inline src/main/scala/me/rexim/issuestant/SplitIssueTree.scala]: https://github.com/puffnfresh/wartremover#defaultarguments

I'm open to discuss that.

File: src/main/scala/me/rexim/issuestant/Main.scala [hunk 2]
```diff
@@ -48,6 +48,7 @@ object Main {
   }
 
   def main(args: Array[String]): Unit = {
+    @SuppressWarnings(Array("org.wartremover.warts.AsInstanceOf"))
```
Comment (ForNeVeR) [inline src/main/scala/me/rexim/issuestant/Main.scala]: I was forced to check your commit comment to learn why do we need that.

I suggest to put a comment above every non-obvious warning suppression statement.

File: src/main/scala/me/rexim/issuestant/SplitIssueTree.scala (hunk 1 above)
Comment (ForNeVeR) [inline src/main/scala/me/rexim/issuestant/SplitIssueTree.scala]: For now the answer satisfies me.

File: src/main/scala/me/rexim/issuestant/Main.scala [hunk 3]
```diff
@@ -48,6 +48,10 @@ object Main {
   }
 
   def main(args: Array[String]): Unit = {
+    // Circe macros expansions use asInstancOf, which makes
```
Comment (ForNeVeR) [inline src/main/scala/me/rexim/issuestant/Main.scala]: `asInstancOf` → `asInstanceOf`

**Comment by rexim** [comment]
@ForNeVeR thank you very much for the review!
